app/controllers/api/web/push_subscriptions_controller.rb

```diff
@@ -3,45 +3,56 @@
 class Api::Web::PushSubscriptionsController < Api::Web::BaseController
   before_action :require_user!
   before_action :set_push_subscription, only: :update
+  before_action :destroy_previous_subscriptions, only: :create, if: :prior_subscriptions?
+  after_action :update_session_with_subscription, only: :create
 
   def create
-    active_session = current_session
+    @push_subscription = ::Web::PushSubscription.create!(web_push_subscription_params)
 
-    unless active_session.web_push_subscription.nil?
-      active_session.web_push_subscription.destroy!
-      active_session.update!(web_push_subscription: nil)
-    end
+    render json: @push_subscription, serializer: REST::WebPushSubscriptionSerializer
+  end
 
-    # Mobile devices do not support regular notifications, so we enable push notifications by default
-    alerts_enabled = active_session.detection.device.mobile? || active_session.detection.device.tablet?
+  def update
+    @push_subscription.update!(data: data_params)
+    render json: @push_subscription, serializer: REST::WebPushSubscriptionSerializer
+  end
 
-    data = {
-      policy: 'all',
-      alerts: Notification::TYPES.index_with { alerts_enabled },
-    }
+  private
 
-    data.deep_merge!(data_params) if params[:data]
+  def active_session
+    @active_session || current_session
```

Comment:
`@active_session` is never used.

I guess the following was meant:
```suggestion
    @active_session ||= current_session
```

But afaik, we need to be cautious with that pattern: https://island94.org/2023/10/writing-object-shape-friendly-code-in-ruby